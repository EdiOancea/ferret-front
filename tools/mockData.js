const company = {
  id: 1,
  name: 'Name',
  address: 'Address',
  business: 'Business',
  rating: 5,
  manager: 'Manager',
  active: true,
};

export { company };
