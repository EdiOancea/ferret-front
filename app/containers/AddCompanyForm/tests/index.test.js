import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import {
  AddCompanyForm,
  mapStateToProps,
  mapDispatchToProps,
} from 'containers/AddCompanyForm';
import { company } from '../../../../tools/mockData';

describe('<AddCompanyForm />', () => {
  it('should show an error if add company failed', () => {
    const addCompanyState = fromJS({
      errorMessage: 'This is an error',
    });
    const initialState = fromJS({
      addCompany: addCompanyState,
    });
    expect(mapStateToProps(initialState).errorMessage).toEqual(
      'This is an error',
    );
  });

  it('should add company when button is clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).onSubmit(company);
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: 'ADD_COMPANY',
      data: company,
    });
  });

  it('should render the add company form correctly', () => {
    const tree = shallow(<AddCompanyForm />);
    expect(tree).toMatchSnapshot();
  });
});
