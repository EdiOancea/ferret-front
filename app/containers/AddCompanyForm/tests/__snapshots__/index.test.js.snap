// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCompanyForm /> should render the add company form correctly 1`] = `
<WithStyles(FormComponent)
  fields={
    Array [
      Object {
        "options": Object {
          "label": "Name",
          "name": "name",
          "type": "name",
        },
        "type": "input",
      },
      Object {
        "options": Object {
          "label": "Address",
          "name": "address",
          "type": "address",
        },
        "type": "input",
      },
      Object {
        "options": Object {
          "label": "Business",
          "name": "business",
          "type": "business",
        },
        "type": "input",
      },
      Object {
        "options": Object {
          "label": "Rating",
          "name": "rating",
          "type": "rating",
        },
        "type": "input",
      },
      Object {
        "options": Object {
          "label": "Manager",
          "name": "manager",
          "type": "manager",
        },
        "type": "input",
      },
      Object {
        "options": Object {
          "label": "Active",
          "name": "active",
          "type": "active",
        },
        "type": "input",
      },
      Object {
        "options": Object {
          "name": "submitButton",
          "text": "SUBMIT",
        },
        "type": "button",
      },
    ]
  }
  initialValues={
    Object {
      "active": "",
      "address": "",
      "business": "",
      "manager": "",
      "name": "",
      "rating": "",
    }
  }
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": undefined,
      "_nodes": Array [
        "active",
        "manager",
        "rating",
        "business",
        "address",
        "name",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "active": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {},
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
        },
        "address": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "business": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "manager": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {},
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
        },
        "name": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
            "undefined": false,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "rating": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {},
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
/>
`;
